### PR TITLE
Zephyr port: add certificate provisioning

### PR DESCRIPTION
**Pull request by szczys**
Add certificate_provisioning example code.

### CODE REVIEW

File: examples/zephyr/certificate_provisioning/Kconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 Golioth, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Golioth application options"
+
+config NET_BUF_TX_COUNT
+	default 32 if GOLIOTH_AUTH_METHOD_CERT && WIFI_ESP_AT
```
Comment (sam-golioth) [inline examples/zephyr/certificate_provisioning/Kconfig:7]: This is already in `examples/zephyr/common/Kconfig.defconfig` so we shouldn't need it here.

File: examples/zephyr/certificate_provisioning/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ESP chip. Flash ESP chip with following command:
+
+```console
+esptool.py write_flash --verify 0x0 PATH_TO_ESP_AT/factory/factory_WROOM-32.bin
+```
+
+Connect nRF52840 DK and ESP32-DevKitC V4 (or other ESP32-WROOM-32 based
+board) using wires:
+
+| nRF52840 DK | ESP32-WROOM-32  | ESP32-WROVER-32 |
```
Comment (sam-golioth) [inline examples/zephyr/certificate_provisioning/README.md:36]: Let's also add the pin mapping for ESP32-C3-MINI-1 as in https://github.com/golioth/golioth-firmware-sdk/pull/165

File: examples/zephyr/certificate_provisioning/README.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+This sample requires that you have:
+
+* A root or intermediate certificate uploaded to the Golioth console
+* A client certificate signed by the private key associated with the
+  root or intermediate certificate
+* The private key associated with the client certificate
+
+For instructions on generating and positioning these certificates, see
+[golioth cert
+auth](https://docs.golioth.io/firmware/zephyr-device-sdk/authentication/certificate-auth).
```
Comment (sam-golioth) [inline examples/zephyr/certificate_provisioning/README.md:100]: Probably want to change this to point to either the Firmware SDK docs or a generic section for generating certificates and adding them to the console.

**Review by sam-golioth — APPROVED** [review]

File: examples/zephyr/certificate_provisioning/README.md (hunk 3 above)
Comment (szczys) [reply examples/zephyr/certificate_provisioning/README.md:100]: I think this link is relevant. It provides instructions for a procedure out-of-scope for the Firmware SDK just like the link to installing mcumgr. I could change the link to point to the scripts directory but there's little explanation there, and no discription of uploading the root cert to Golioth.

Comment (sam-golioth) [reply examples/zephyr/certificate_provisioning/README.md:100]: I agree it's relevant, but right now it points to _Zephyr_ SDK documentation from the CP SDK, which is confusing.

Comment (szczys) [reply examples/zephyr/certificate_provisioning/README.md]: Ah, sorry. I didn't realize the issue was the wrong docs section.

We already have a cert provision

Comment (sam-golioth) [reply examples/zephyr/certificate_provisioning/README.md]: Yup, that's good with me!

**Pull request merged**